ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -149,11 +306,107 @@ private static Certificate readX509Certificate(Path certificateFilePath) {
+            String certPemFileString =
+                    privateKeyFilePath.equals(certFilePath) ? keyPemFileString : readFileAsString(certFilePath);
+            certificates = getCertificatesFromString(certPemFileString);
+        } catch (IOException | GeneralSecurityException e) {
+            throw new RuntimeException(String.format("Failed to read certificates from file at \"%s\"",
+                    certFilePath), e);
+        }
+
+        return new KeyStore.PrivateKeyEntry(privateKey, certificates);
+    }
+
+    private static String readFileAsString(Path path) throws IOException {
+        byte[] fileBytes = Files.readAllBytes(path);
+        return new String(fileBytes, Charsets.UTF_8);
+    }
+
+    /**
+     * Returns a PrivateKey representing the first PEM formatted RSA key in the provided string. Throws an exception if
+     * the provided string does not contain a valid RSA key.
+     *
+     * @param pemFileString
+     *        string that contains PEM formatted RSA key (PKCS#1). The string can contain other content as well (for
+     *        example, certificates or other information) as long as properly formatted RSA key content exists in the
+     *        string.
+     * @return PrivateKey representing the first RSA key in the provided string
+     */
+    private static PrivateKey getPrivateKeyFromString(String pemFileString) throws IOException,
+            GeneralSecurityException {
+        Matcher matcher = KEY_PATTERN.matcher(pemFileString);
+        if (!matcher.find()) {
+            throw new GeneralSecurityException(
+                    String.format("unable to find valid RSA key in the provided string: %s", pemFileString));
+        }
+
+        // get content between headers and strip newlines to get Base64 encoded ASN1 DER only
+        String privateKeyString = matcher.group(1).replace("\n", "");
+
+        // read private key
+        RSAPrivateKeySpec rsaPrivKeySpec = Pkcs1Readers.readPrivateKey(privateKeyString);
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        return kf.generatePrivate(rsaPrivKeySpec);
+    }
+
+    /**
+     * Returns an array of certificates representing the PEM formatted X.509 certificates in the provided string. The
+     * order of the certificates in the array will match the order that the certificates occur in the string.
+     *
+     * @param pemFileString
+     *        string that contains PEM formatted X.509 certificates. The string can contain other content as well (for
+     *        example, RSA keys or other information) as long as properly formatted certificate content exists in the
+     *        string.
+     * @return array of Certificates that represent the X.509 certificates that were found in the input string in the
+     *         order that they appeared in the string. Will be empty if no certificates were found in the input string.
+     */
+    private static Certificate[] getCertificatesFromString(String pemFileString) throws IOException,
```

Comment:
would prefer to return List<> and turn into [] when used in the `KeyStore.PrivateKeyEntry` constructor